api/utils/helpers.py

```diff
@@ -1,9 +1,91 @@
-from flask import json, Response, request
+import json
+import os
+import logging
+from types import GeneratorType
+
 from sqlalchemy_utils import sort_query
 
 
+def responsify(body, code, headers=None):
+    default_headers = {"Access-Control-Allow-Origin": "*", "Access-Control-Allow-Credentials": True}
+    default_headers.update(headers) if headers else None
+    results_obj = {}
+    if isinstance(body, dict):
+        results_obj = body
+    elif isinstance(body, list) or isinstance(body, str):
+        results_obj = {"data": body}
+    elif isinstance(body, GeneratorType):
+        results_obj = {"data": [item for item in body]}  # Unpack generator
+    return {
+        "statusCode": code,
+        "isBase64Encoded": False,
+        "headers": default_headers,
+        "body": json.dumps(results_obj)
+    }
+
+
+class Abort(Exception):
+    def __init__(self, code, reason):
+        msg = f"ERROR:{code} MSG: {reason}"
+        super(Abort, self).__init__(msg)
+
+
+class Serverless:
+    logger = logging.getLogger()
+    logger.setLevel(logging.INFO)
+    args = {}
+
+    class Args(object):
+        def __init__(self, parameters={}):
+            self.params = parameters
+
+        def get(self, key, default, type=str):
+            try:
+                value = self.params.get(key)
+                if value:
+                    return type(value)
+                else:
+                    return default
+            except TypeError:
+                raise Abort(400, f"{type} is not a valid Type")
+            except ValueError:
+                return Abort(400, f"Invalid literal {key} for type {type}")
+
+    @classmethod
+    def lambda_handler(cls, event):
+        cls.logger.info('## ENVIRONMENT VARIABLES')
+        cls.logger.info(os.environ)
+        cls.logger.info('## EVENT')
+        cls.logger.info(event)
+
+    @classmethod
+    def route(cls, func):
+        def wrapper(event: dict, _):
+            cls.lambda_handler(event)
+            stringParams: dict = event.get('queryStringParameters') or {}
+            pathParams: dict = event.get('pathParameters') or {}
+            cls.args = cls.Args(stringParams)
+
+            cls.logger.info(f"Arguments {stringParams} | {pathParams}")
+
+            try:
+                body, *context = func(pathParams) if pathParams else func()
```

Comment:
In Flask, it casted path parameters based on a converter i.e. `<int:term_id>` passed term_id into the function as an integer. I think right now the URL path parameters are being passed in as strings. Is there a way to define their types as ints or strings (if we wanted them as strings) in the serverless config?

If that's not possible, you could probably get away with just casting all the url params to ints (and if there's an error on the conversion, leave it as a string).

Reply:
it returns them as strings, it doesn't automatically cast. Casting everything is a bad design decision as well, you shouldn't be using errors as conditionals/control-flow. If you need an int, you should manually cast that specific param.